### DISCUSSION TITLE
Is persistence for nodes necessary or can it be omitted in a gitops-case?

### DISCUSSION THREAD

**Discussion by joggeli34** [emqx/emqx#11591] [Q&A]
We are running a cluster with kubernetes and are using gitops for deployment. Therefore, we don't change the configuration through the web-ui, only by changing the helm configuration.

We decided to use no persistence for the nodes, as this simplifies the deployment. Additionally, if there is a problem with the cluster (split or something else) each pod can be deleted simultaneously and with only a short break, the cluster is started with completely new configuration without loading the probably broken state.

I know that when a node is killed unexpectedly, we could probably lose some messages, but that is a risk we are willing to take. But if the nodes are restarted for updating or node balancing, nothing should should be lost.

I wonder if there are any risks or downsides of this assumptions and it is ok to run the nodes without persistence or if there is a good reason to add the persistence.

**Comment by qzhuyan** [comment]
EMQX is stateful application. I won't suggest you to use no persistence because there are other states (such as cluster info) are persistented.

But depends on your application, as you said you could bear message losts, bring up fresh cluster without any states, that is fine.
